src/treq/client.py

```diff
@@ -304,19 +314,41 @@ def _convert_files(files):
         yield (param, (file_name, content_type, IBodyProducer(fobj)))
 
 
-def _combine_query_params(url, params):
-    parsed_url = urlparse(url.encode('ascii'))
-
-    qs = []
+def _coerced_query_params(params):
+    """
+    Carefully coerce *params* in the same way as `urllib.parse.urlencode()`
 
-    if parsed_url.query:
-        qs.extend([parsed_url.query, b'&'])
+    Parameter names and values are coerced to unicode. As a special case,
+    `bytes` are decoded as ASCII.
```

Comment:
(Again, why ASCII rather than latin-1?)

Reply:
This matches what the `urllib.parse` functions did. TBH I'd rather be stricter about types than this, it's just here for compatibility with the previous implementation. Switching to `charmap` may make sense, but I'd want to add additional test cases to formalize the expected behavior.